docs/developer_guide.md

```diff
@@ -104,6 +104,36 @@ opensearch-data-prepper:1.2.0-SNAPSHOT
 ```
 
 
+## Contributing your Code to Data Prepper
+
+When you wish to contribute a change to Data Prepper, please create a GitHub Pull Request (PR). You should create your PR to target `main`.
```

Comment:
Take it or leave it: `When you wish` sounds passive and doesn't acknowledge our guidelines or standards. I would recommend using `ready`. I would also avoid the use of create in the second sentence as it is redundant from the first sentence and the requirement isn't on creating but rather the PR itself. 

```
When you are ready to contribute a change to Data Prepper, please create a GitHub Pull Request (PR). Your PR should target `main`.
```